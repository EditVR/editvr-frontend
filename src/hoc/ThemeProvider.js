/**
 * @file ThemeProvider.js
 * Exports a HOC ThemeProvider component that attaches children their MUI theme.
 */

import React from 'react';
import { MuiThemeProvider } from '@material-ui/core';

import muiTheme from '../styles/muiTheme';

const ThemeProvider = WrappedComponent => props => (
  <MuiThemeProvider theme={muiTheme}>
    <WrappedComponent {...props} />
  </MuiThemeProvider>
);

export default ThemeProvider;
