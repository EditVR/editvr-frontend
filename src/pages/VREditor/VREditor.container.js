/**
 * @file VREditor.container.js
 * Exports a redux-connected VREditor component.
 */

import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import VREditor from './VREditor';

const mapDispatchToProps = dispatch => ({
  dispatch
});

const mapState = ({ openExperience, user, component }) => ({
  experience: openExperience,
  component,
  user
});

export default withRouter(connect(mapState, mapDispatchToProps)(VREditor));
