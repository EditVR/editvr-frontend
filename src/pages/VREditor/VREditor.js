/**
 * @file VREditor.js
 * Exports a React component that render's EditVR's VREditor interface.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { withStyles, Grid, Typography, Button } from '@material-ui/core';
import { AddBox } from '@material-ui/icons';

import { SceneCards } from '../../components';
import {
  OPEN_EXPERIENCE_FETCH_FOR_USER,
  MODE_SCENE_CREATE
} from '../../constants';
import { VREditorLayout } from '../../layouts';
import VREditorStyles from './VREditor.style';

class VREditor extends Component {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    classes: PropTypes.shape({
      columnRight: PropTypes.string.isRequired,
      columnLeft: PropTypes.string.isRequired,
      button: PropTypes.string.isRequired,
      buttonIcon: PropTypes.string.isRequired
    }).isRequired,
    user: PropTypes.shape({
      uid: PropTypes.string.isRequired,
      authentication: PropTypes.shape({
        accessToken: PropTypes.string.isRequired,
        csrfToken: PropTypes.string.isRequired
      }).isRequired
    }).isRequired,
    experience: PropTypes.shape({
      item: PropTypes.shape({
        title: PropTypes.string
      })
    }),
    match: PropTypes.shape({
      params: PropTypes.shape({
        experienceSlug: PropTypes.string.isRequired
      }).isRequired
    }).isRequired
  };

  static defaultProps = {
    experience: {
      item: {
        title: false
      }
    }
  };

  /**
   * {@inheritdoc}
   */
  componentWillMount() {
    const {
      dispatch,
      user,
      match: {
        params: { experienceSlug }
      }
    } = this.props;
    dispatch({
      type: OPEN_EXPERIENCE_FETCH_FOR_USER,
      experienceSlug,
      user
    });
  }

  /**
   * {@inheritdoc}
   */
  render() {
    const {
      match: {
        params: { editorMode }
      },
      experience: { item: experience },
      classes
    } = this.props;
    const {
      title,
      field_experience_path: path,
      field_scenes: scenes
    } = experience;

    return (
      <VREditorLayout
        title={experience.title ? `Editing ${title} Experience` : 'Loading...'}
        leftAside={
          <Grid container align="stretch">
            <Grid
              item
              xs={3}
              id="column--left_aside"
              className={classes.columnLeft}
            />
            <Grid
              item
              xs={9}
              id="column--left_aside"
              className={classes.columnRight}
            >
              <Typography variant="title">Scenes</Typography>
              {scenes && (
                <Button
                  variant="raised"
                  color="primary"
                  component={Link}
                  className={classes.button}
                  to={`/experience/vreditor/${path}/scene/${MODE_SCENE_CREATE}`}
                >
                  Create
                  <AddBox className={classes.buttonIcon} />
                </Button>
              )}
              <SceneCards />
            </Grid>
          </Grid>
        }
        rightAside="Right Sidebar"
      >
        {editorMode === MODE_SCENE_CREATE && (
          <Typography>Create a scene</Typography>
        )}
      </VREditorLayout>
    );
  }
}

export default withStyles(VREditorStyles)(VREditor);
