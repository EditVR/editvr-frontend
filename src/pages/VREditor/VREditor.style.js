/**
 * @file VREditor.style.js
 * Exports VREditor component styles.
 */

export default theme => ({
  columnRight: {
    padding: theme.spacing.unit * 2,
    height: 'calc(100vh - 64px)',
    maxHeight: 'calc(100vh - 64px)',
    userSelect: 'none',
    overflowY: 'scroll'
  },
  columnLeft: {
    padding: theme.spacing.unit * 2,
    userSelect: 'none'
  },
  button: {
    marginTop: theme.spacing.unit * 2,
    marginBottom: theme.spacing.unit * 2
  },
  buttonIcon: {
    marginLeft: theme.spacing.unit
  }
});
