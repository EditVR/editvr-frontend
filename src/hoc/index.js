/**
 * @file index.js
 * Exports all hoc components.
 */

import ThemeProvider from './ThemeProvider';

export { ThemeProvider };
