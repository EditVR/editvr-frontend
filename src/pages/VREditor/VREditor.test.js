/**
 * @file VREditor.test.js
 * Contains tests for VREditor.js.
 */

import React from 'react';
import configureStore from 'redux-mock-store';
import { createMount } from '@material-ui/core/test-utils';
import { Provider } from 'react-redux';
import { MemoryRouter as Router, Route } from 'react-router-dom';

import { ThemeProvider } from '../../hoc';
import VREditor from './VREditor.container';

const ThemedVREditor = ThemeProvider(VREditor);

describe('<VREditor />', () => {
  it('Matches its snapshot', () => {
    const store = configureStore()({
      loadingBar: {
        default: 0
      },
      user: {
        username: 'bender',
        uid: '10',
        authentication: {
          accessToken: 'token',
          csrfToken: 'token'
        }
      },
      openExperience: {
        item: {
          title: 'test'
        }
      }
    });
    expect(
      createMount()(
        <Provider store={store}>
          <Router initialEntries={['/experience/vreditor/test']}>
            <Route
              path="/experience/vreditor/:experienceSlug"
              component={ThemedVREditor}
            />
          </Router>
        </Provider>
      ).html()
    ).toMatchSnapshot();
  });
});
