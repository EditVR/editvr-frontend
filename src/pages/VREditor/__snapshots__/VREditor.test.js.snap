// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VREditor /> Matches its snapshot 1`] = `
<div
  align="stretch"
  className="MuiGrid-container-6 VREditorLayout-wrapper-1"
  id="editor__wrapper"
>
  <div />
  <header
    className="MuiPaper-root-105 MuiPaper-elevation4-111 MuiAppBar-root-97 MuiAppBar-positionFixed-98 MuiAppBar-colorPrimary-103 mui-fixed VREditorLayout-appBar-4"
  >
    <div
      className="MuiToolbar-root-132 MuiToolbar-gutters-133"
    >
      <a
        aria-label="Back to Dashboard"
        className="MuiButtonBase-root-140 MuiIconButton-root-134 MuiIconButton-colorInherit-135"
        href="/dashboard"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiIconButton-label-139"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-143"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-149"
        />
      </a>
      <img
        alt="EditVR logo"
        className="VREditorLayout-logo-5"
        src="editvr-logo.svg"
      />
      <h2
        className="MuiTypography-root-156 MuiTypography-title-162"
      >
        Edit Example Experience
      </h2>
    </div>
  </header>
  <div
    className="MuiGrid-item-7 MuiGrid-grid-xs-3-35 VREditorLayout-aside-2"
  >
    Left Sidebar
  </div>
  <div
    className="MuiGrid-item-7 MuiGrid-grid-xs-6-38 VREditorLayout-middle-3"
  >
    Hey there, this will soon be a VR scene.
  </div>
  <div
    className="MuiGrid-item-7 MuiGrid-grid-xs-3-35 VREditorLayout-aside-2"
  >
    Right Sidebar
  </div>
</div>
`;
